python/claude/sample-agent/agent.py

```diff
@@ -0,0 +1,502 @@
+    # INITIALIZATION
+    # =========================================================================
+    # <Initialization>
+
+    def __init__(self):
+        """Initialize the Claude agent."""
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+        # Initialize authentication options
+        self.auth_options = LocalAuthenticationOptions.from_environment()
+
+        # Create Claude client
+        self._create_client()
+
+        # Claude client instance (will be set per conversation)
+        self.client: ClaudeSDKClient | None = None
+
+    # </Initialization>
+
+    # =========================================================================
+    # CLIENT CREATION
+    # =========================================================================
+    # <ClientCreation>
+
+    def _create_client(self):
+        """Create the Claude Agent SDK client options"""
+        # Get model from environment or use default
+        model = os.getenv("CLAUDE_MODEL", "claude-sonnet-4-20250514")
+        
+      
+        # Get API key
+        api_key = os.getenv("ANTHROPIC_API_KEY")
+        if not api_key:
+         raise EnvironmentError("Missing ANTHROPIC_API_KEY. Please set it before running.")
+
+        # Configure Claude options
+        self.claude_options = ClaudeAgentOptions(
+            model=model,
+            # Enable extended thinking for detailed reasoning
+            max_thinking_tokens=1024,
+            # Allow web search and basic file operations
+            allowed_tools=["WebSearch", "Read", "Write", "WebFetch"],
+            # Auto-accept edits for smoother operation
+            permission_mode="acceptEdits",
+            continue_conversation=True
+        )
+
+        logger.info(f"✅ Claude Agent configured with model: {model}")
+
+
+
+
+    # </ClientCreation>
+
+    # =========================================================================
+    # INITIALIZATION AND MESSAGE PROCESSING
+    # =========================================================================
+    # <MessageProcessing>
+
+    async def initialize(self):
+        """Initialize the agent"""
+        logger.info("Initializing Claude Agent...")
+        try:
+            logger.info("Claude Agent initialized successfully")
+        except Exception as e:
+            logger.error(f"Failed to initialize agent: {e}")
+            raise
+
+    async def process_user_message(
+        self, message: str, auth: Authorization, context: TurnContext
+    ) -> str:
+        """Process user message using the Claude Agent SDK with observability tracing"""
+        
+        # Check if observability is enabled
+        enable_observability = os.getenv("ENABLE_OBSERVABILITY", "false").lower() in ("true", "1", "yes")
+        
+        # Create observability objects if available and enabled
+        invoke_scope = None
+        baggage_context = None
+        if OBSERVABILITY_AVAILABLE and enable_observability:
+            try:
```

Comment:
All these logic could be cleaned up removing the condition around observability, moving some of the logic to functions and using `with ...` instead of `scope.__enter__()`.